tests/ballerina-integration-test/src/test/java/org/ballerinalang/test/service/http2/Http2ToHttp1FallbackTestCase.java

```diff
@@ -38,4 +39,15 @@ public void testFallback() throws IOException {
         String responseData = response.getData();
         Assert.assertTrue(responseData.contains("1.1"), responseData);
     }
+
+    @Test(description = "Test HTTP/2.0 to HTTP/1.1 server fallback scenario with SSL enabled")
+    public void testFallbackWithSSL() throws IOException {
+        String serviceUrl = "https://localhost:9096/hello";
+        String serverHome = serverInstance.getServerHome();
+        HttpResponse response = HttpsClientRequest.doGet(serviceUrl, serverHome);
+        Assert.assertNotNull(response);
+        Assert.assertEquals(response.getResponseCode(), 200, "Response code mismatched");
+        String responseData = response.getData();
+        Assert.assertTrue(responseData.contains("1.1"), responseData);
```

Comment:
Here why the message is whatever the data sent with the response? Can we guarantee a useful message will be returned in case this assertion fails?

Reply:
Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/10951/commits/34db2e3f56d52290a4edc2aa1ef4f7575750d2b1